crates/symbolicator/src/services/objects/meta_cache.rs

```diff
@@ -103,47 +96,57 @@ impl CacheItemRequest for FetchFileMetaRequest {
     /// This returns [`CacheStatus::CacheSpecificError`] if the download failed.  If the
     /// data cache is [`CacheStatus::Negative`] or [`CacheStatus::Malformed`] then the same
     /// status is returned.
-    fn compute(&self, path: &Path) -> BoxFuture<'static, Result<CacheStatus, Self::Error>> {
+    ///
+    /// This is the actual implementation of [`CacheItemRequest::compute`] for
+    /// [`FetchFileMetaRequest`] but outside of the trait so it can be written as async/await
+    /// code.
+    async fn compute_file_meta(self, path: PathBuf) -> Result<CacheStatus, ObjectError> {
```

Comment:
this version is much nicer than some of the others because of the `self` arg which means it doesn't need a really long arg list.  I guess doing this for the others is too much refactoring (probably not all `self`s are clonable)?